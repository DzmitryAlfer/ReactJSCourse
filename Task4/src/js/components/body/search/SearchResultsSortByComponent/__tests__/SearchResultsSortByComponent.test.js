import React from 'react';
import Renderer from 'react-test-renderer';
import {SearchResultsSortByComponent, SearchResultsSortByComponentState} from '../index'
import {mount} from 'enzyme';

describe('SearchResultsSortByComponent', () => {
    test('Component render1', () => {

        const tree = Renderer.create(
            <SearchResultsSortByComponent/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Test button click', () => {
        const onButtonClick = jest.fn();
        const wrapper = mount((
            <SearchResultsSortByComponent onSortBy={onButtonClick} />
        ));

        wrapper.find('.rating').simulate('click');
        expect(onButtonClick).toHaveBeenCalledWith(SearchResultsSortByComponentState.RATING);

        wrapper.find('.release').simulate('click');
        expect(onButtonClick).toHaveBeenCalledWith(SearchResultsSortByComponentState.RELEASE_DATE);
    });
});