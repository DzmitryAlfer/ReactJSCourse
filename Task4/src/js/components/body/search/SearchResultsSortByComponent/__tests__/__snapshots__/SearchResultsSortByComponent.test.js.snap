// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsSortByComponent Component render1 1`] = `
<div
  className="search-info_item"
>
  <span
    className="search-info_text"
  >
    Sort by
  </span>
  <button
    className="release sort-by_button sort-by_button--selected"
    onClick={[Function]}
  >
    release date
  </button>
  <button
    className="rating sort-by_button"
    onClick={[Function]}
  >
    rating
  </button>
</div>
`;
